v3/errors.go

```diff
@@ -0,0 +1,127 @@
+	http.StatusForbidden:                     ErrForbidden,
+	http.StatusNotFound:                      ErrNotFound,
+	http.StatusMethodNotAllowed:              ErrMethodNotAllowed,
+	http.StatusNotAcceptable:                 ErrNotAcceptable,
+	http.StatusProxyAuthRequired:             ErrProxyAuthRequired,
+	http.StatusRequestTimeout:                ErrRequestTimeout,
+	http.StatusConflict:                      ErrConflict,
+	http.StatusGone:                          ErrGone,
+	http.StatusLengthRequired:                ErrLengthRequired,
+	http.StatusPreconditionFailed:            ErrPreconditionFailed,
+	http.StatusRequestEntityTooLarge:         ErrRequestEntityTooLarge,
+	http.StatusRequestURITooLong:             ErrRequestURITooLong,
+	http.StatusUnsupportedMediaType:          ErrUnsupportedMediaType,
+	http.StatusRequestedRangeNotSatisfiable:  ErrRequestedRangeNotSatisfiable,
+	http.StatusExpectationFailed:             ErrExpectationFailed,
+	http.StatusTeapot:                        ErrTeapot,
+	http.StatusMisdirectedRequest:            ErrMisdirectedRequest,
+	http.StatusUnprocessableEntity:           ErrUnprocessableEntity,
+	http.StatusLocked:                        ErrLocked,
+	http.StatusFailedDependency:              ErrFailedDependency,
+	http.StatusTooEarly:                      ErrTooEarly,
+	http.StatusUpgradeRequired:               ErrUpgradeRequired,
+	http.StatusPreconditionRequired:          ErrPreconditionRequired,
+	http.StatusTooManyRequests:               ErrTooManyRequests,
+	http.StatusRequestHeaderFieldsTooLarge:   ErrRequestHeaderFieldsTooLarge,
+	http.StatusUnavailableForLegalReasons:    ErrUnavailableForLegalReasons,
+	http.StatusInternalServerError:           ErrInternalServerError,
+	http.StatusNotImplemented:                ErrNotImplemented,
+	http.StatusBadGateway:                    ErrBadGateway,
+	http.StatusServiceUnavailable:            ErrServiceUnavailable,
+	http.StatusGatewayTimeout:                ErrGatewayTimeout,
+	http.StatusHTTPVersionNotSupported:       ErrHTTPVersionNotSupported,
+	http.StatusVariantAlsoNegotiates:         ErrVariantAlsoNegotiates,
+	http.StatusInsufficientStorage:           ErrInsufficientStorage,
+	http.StatusLoopDetected:                  ErrLoopDetected,
+	http.StatusNotExtended:                   ErrNotExtended,
+	http.StatusNetworkAuthenticationRequired: ErrNetworkAuthenticationRequired,
+}
+
+func handleHTTPErrorResp(resp *http.Response) error {
+	if resp.StatusCode >= 400 && resp.StatusCode <= 599 {
+		var res struct {
+			Message string `json:"message"`
+		}
+
+		data, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return fmt.Errorf("error reading response body: %s", err)
+		}
+
+		if json.Valid(data) {
+			if err = json.Unmarshal(data, &res); err != nil {
+				return fmt.Errorf("error unmarshaling response: %s", err)
+			}
+		} else {
+			res.Message = string(data)
+		}
+
+		err, ok := httpStatusCodeErrors[resp.StatusCode]
+		if ok {
+			if err == ErrNotFound {
```

Comment:
Why do we need this special case?

Reply:
I reproduced the past behavior, to not ship the message (probably no message for 404 that why) in the error for 404 but I think I can test to do like other codes, yes

Reply:
I removed it message is here and more explicit then, thanks @sauterp